woocommerce/compatibility/class-sv-wc-order-compatibility.php

```diff
@@ -718,6 +718,42 @@ public static function delete_order_meta( $order, string $meta_key, $meta_value
 	}
 
 
+	/**
+	 * Get global $post or $theorder order object according to HPOS availability.
+	 *
+	 * @since x.y.z
+	 * @return \WC_Order|\WP_Post  object
+	 */
+	public static function get_post_or_order() {
+
+		global $post, $theorder;
+
+		if ( ! is_null( $theorder ) && static::is_order( $theorder ) ) {
+			return $theorder;
+		} else if( ! is_null( $post ) ) {
+			return $post;
+		}
+	
+	}
+
+	/**
+	 * Get global $post or $theorder order status.
+	 *
+	 * TODO: Refactor this method to get order status by order object or ID. If object order or ID does not pass in params then get global order status.
+	 * @since x.y.z
+	 * @return string order status
+	 */
+	public static function get_order_status() {
+
+	$post_or_order_object	= static::get_post_or_order();
+
+		if ( $post_or_order_object instanceof \WC_Order ) {
+			return $post_or_order_object->get_status();
+		} elseif ( $post_or_order_object instanceof \WP_Post ) {
+			return get_post_status( $post_or_order_object );
+		}
+	}
+
```

Comment:
I am also not sure if this method is needed either.

Also this method does not account for the possibility neither global is set and will return void.

But also, I believe if the purpose of adding this is solely to fix the logic in the hook callback below, then we could just move the corresponding logic in the callback. I made a quick search on plugins and I don't think there are more instances of `admin_footer-edit.php`. Sure enough, there could be more contexts where we need detection of an order edit screen - but in that case we could benefit from a different helper method.